gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationsResource.java

```diff
@@ -167,11 +173,22 @@ public Response getApplications(
                 RolePermissionAction.CREATE
             );
         } else {
-            applicationIds = applicationService.findIdsByUser(
+            ApplicationQuery applicationQuery = ApplicationQuery.builder()
+                .user(getAuthenticatedUser())
+                .status(ApplicationStatus.ACTIVE.name())
+                .build();
+            Page<ApplicationListItem> applicationPage = applicationService.search(
                 executionContext,
-                getAuthenticatedUser(),
-                applicationsOrderParam.toSortable()
+                applicationQuery,
+                applicationsOrderParam.toSortable(),
+                new PageableImpl(paginationParam.getPage(), paginationParam.getSize())
             );
+
+            applicationIds = applicationPage.getContent().stream().map(ApplicationListItem::getId).collect(Collectors.toList());
+            Map<String, Object> paginateMeta = new HashMap<>();
+            paginateMeta.put("totalElements", applicationPage.getTotalElements());
+            resPageMetaData.put("paginateMetaData", paginateMeta);
+            withPagination = false;
```

Comment:
so it's paginated i'm right ? before pagination are done by `createListResponse` now it's don here ?